js/filter.js

```diff
@@ -0,0 +1,118 @@
+const mapFiltersElement = document.querySelector('.map__filters');
+const housingTypeElement = mapFiltersElement.querySelector('#housing-type');
+const housingPriceElement = mapFiltersElement.querySelector('#housing-price');
+const housingRoomsElement = mapFiltersElement.querySelector('#housing-rooms');
+const housingGuestsElement = mapFiltersElement.querySelector('#housing-guests');
+const featuresElements = mapFiltersElement.querySelectorAll('input[name="features"]');
+
+const getFilteredPrice = (ads, property, filter) => {
+  let filteredList = [];
+
+  switch(filter) {
+    case 'any':
+      filteredList = ads;
+      break;
+    case 'middle':
+      filteredList = ads.filter((ad) => {
+        if (10000 <= ad.offer[property] && ad.offer[property] <= 50000) {
+          return ad;
+        }
+      });
+      break;
+    case 'low':
+      filteredList = ads.filter((ad) => {
+        if (ad.offer[property] < 10000) {
+          return ad;
+        }
+      });
+      break;
+    case 'high':
+      filteredList = ads.filter((ad) => {
+        if (ad.offer[property] > 50000) {
+          return ad;
+        }
+      });
+      break;
+  }
+
+  return filteredList;
+};
+
+const getFilteredAttributes = (ads, property, filter) => {
+  const filteredList = ads.filter((ad) => {
+    if (ad.offer[property].toString() === filter || filter === 'any') {
+      return ad;
+    }
+  });
+
+  return filteredList;
+};
+
+const getFilteredFeatures = (ads, property, filter, isChecked = false) => {
+  let filteredList = [];
+
+  ads.forEach((ad) => {
+    if (isChecked) {
+      ad.offer.features.forEach((feature) => {
+        if (feature === filter) {
+          filteredList.push(ad);
+        }
+      });
+    } else {
+      filteredList.push(ad);
+    }
+  });
+
+  return filteredList;
+};
+
+const setTypeFilter = (cb) => {
+  housingTypeElement.addEventListener('change', () => {
+    cb();
+  });
```

Comment:
```suggestion
  housingTypeElement.addEventListener('change', cb);
```
абсолютно аналогично по логике, но короче запись